### ISSUE TITLE
Translations for config flows

### ISSUE THREAD

**Issue by balloob** [home-assistant/home-assistant-polymer#1019]
_Creating it in this repo but it is a hybrid about frontend fetching backend translations._

When we load translations from the backend, we only load the strings for the current loaded components. This however won't work for the translation strings for the config flows to create a config entry. The component will only be set up once a config flow is finished.

Without `config_entry_example:` and `hue:` in `configuration.yaml`:

![image](https://user-images.githubusercontent.com/1444314/37787408-129ea652-2dbc-11e8-9b90-e7cbadadbb6e.png)

With `config_entry_example:` and `hue:` in `configuration.yaml`:

![image](https://user-images.githubusercontent.com/1444314/37787100-750bf246-2dbb-11e8-89f2-4791d4f35c4b.png)

Reported by @Kane610 
CC @armills

**Comment by balloob** [comment]
Created a pull request with the temporary fix: https://github.com/home-assistant/home-assistant/pull/13394

**Comment by iantrich** [comment]
Anything beyond the temp fix needed for this?

**Comment by armills** [comment]
I think we can mark this closed, since the issue is fixed now. If we want to tackle optimization we can open a separate issue for that.
